sql-plugin/src/main/340/scala/com/nvidia/spark/rapids/shims/SparkBaseShim.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import ai.rapids.cudf.DType
+import com.nvidia.spark.rapids.{DecimalUtil, ExprChecks, ExprMeta, ExprRule, GpuCheckOverflow, GpuExpression, SparkShims, TypeSig}
+import com.nvidia.spark.rapids.GpuOverrides.expr
+
+import org.apache.spark.sql.catalyst.expressions.{CheckOverflow, Divide, Expression, Multiply}
+import org.apache.spark.sql.rapids.{GpuDecimalDivide, GpuDecimalMultiply}
+
+/**
+ * Base class of every Shim
+ */
+trait SparkBaseShim extends SparkShims {
+  override def getExprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = {
+    val exprs: Map[Class[_ <: Expression], ExprRule[_ <: Expression]] = Seq(
+      expr[CheckOverflow](
+        "CheckOverflow after arithmetic operations between DecimalType data",
+        ExprChecks.unaryProjectInputMatchesOutput(TypeSig.DECIMAL_128,
+          TypeSig.DECIMAL_128),
+        (a, conf, p, r) => new ExprMeta[CheckOverflow](a, conf, p, r) {
+          private[this] lazy val binExpr = childExprs.head
+          private[this] lazy val lhs = binExpr.childExprs.head
```

Comment:
@revans2 I am not sure if I have done this correctly. I am assuming the types being set are the original types so there is no need to run the method that tries to guess the original types.